test/eoftest/eoftest_runner.cpp

```diff
@@ -0,0 +1,78 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2022 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "../utils/utils.hpp"
+#include "eoftest.hpp"
+#include <evmc/evmc.hpp>
+#include <evmone/eof.hpp>
+#include <gtest/gtest.h>
+#include <nlohmann/json.hpp>
+
+namespace json = nlohmann;
+
+namespace
+{
+struct EOFValidationTest
+{
+    struct Case
+    {
+        struct Expectation
+        {
+            evmc_revision rev;
+            bool result;
+        };
+        std::string name;
+        evmc::bytes code;
+        std::vector<Expectation> expectations;
+    };
+    std::unordered_map<std::string, Case> cases;
+};
+
+void from_json(const json::json& j, EOFValidationTest::Case& o)
+{
+    const std::optional<evmc::bytes> hex_str{evmc::from_hex(j.at("code").get<std::string>())};
+    if (!hex_str)
+        throw std::invalid_argument{"invalid hex string"};
+    const evmc::bytes& code{hex_str.value()};
```

Comment:
```suggestion
    o.code = *opt_code;
```
1. Taking returned value by reference is always risky. This sometimes works because of the lifetime extension but there are some gotchas leading to dangling pointers. So it should be `const evmc::bytes code{hex_str.value()};`.
2. At this point we know that the optional has a value so we can use check-less `*opt_code` operator.
3. Just avoid going through a temporary `code` variable, assign directly to `o.code`.